### PR TITLE
Allow to configure, if order should switch to backorder

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Allow to configure, if order should switch to backorder. Autoupgrade PR for https://github.com/PrestaShop/PrestaShop/pull/29753.
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| How to test?      | -
| Possible impacts? | -

### CODE REVIEW

File: upgrade/sql/8.1.0.sql [hunk 1]
```diff
@@ -3,3 +3,7 @@ SET NAMES 'utf8mb4';
 
 UPDATE `PREFIX_configuration` SET `value` = 'US/Pacific' WHERE `name` = 'PS_TIMEZONE' AND `value` = 'US/Pacific-New' LIMIT 1;
 DELETE FROM `PREFIX_timezone` WHERE `name` = 'US/Pacific-New';
+
+/* Default configuration for backorder, in order to keep behavior */
+INSERT INTO `PREFIX_configuration` (`name`, `value`, `date_add`, `date_upd`) VALUES
```
Comment (nicosomb) [inline upgrade/sql/8.1.0.sql:8]: Don't we need to add it on installation script too?

**Review by slumericanB — APPROVED** [review]

File: upgrade/sql/8.1.0.sql [hunk 2]
```diff
@@ -3,3 +3,7 @@ SET NAMES 'utf8mb4';
...
+  ('PS_ENABLE_BACKORDER_STATUS', '1', NOW(), NOW());
```
Comment (okom3pom) [inline upgrade/sql/8.1.0.sql:9]: 0 In installation script ?

Comment (Hlavtox) [reply upgrade/sql/8.1.0.sql:9]: Yes, OFF for new installs, ON for upgraded installs, so there are no surprises for current merchants. 👍

Comment (okom3pom) [reply upgrade/sql/8.1.0.sql:9]: OK I agree but there is a conflict

Comment (Hlavtox) [reply upgrade/sql/8.1.0.sql:12]: rebased ;-)

**Review by okom3pom — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by sallemiines — COMMENTED** [review]
Hello @Hlavtox, 

Thanx for the PR ! 

- Auto-upgrade module installation : **OK** 
- Upgrade shop with `PS1.7.8.7` to `develop` (`PS8.1.0`) : **OK** 
- Check that the switch button to backorder is exist in orders settings page : **OK** 
- In FO, add an order with a product out of stock > check that in BO the order status : **backorder** is exist : **OK** 
- Rollback to `PS1.7.8.7 `**OK** 

=> LGTM ! QA :heavy_check_mark: , you can see the attached screen record : 

[Untitled_ Nov 22 2022 12_37 PM.webm](https://user-images.githubusercontent.com/31663369/203305499-e06eda4b-7e6c-4875-b992-956779649222.webm)

Thank you ! :rocket:

**Comment by matks** [comment]
Thank you @Hlavtox

**Pull request merged**